soc/arm/st_stm32/stm32f1/soc.h

```diff
@@ -64,6 +64,10 @@
 #include <stm32f1xx_ll_adc.h>
 #endif
 
+#ifdef CONFIG_DAC_STM32
```

Comment:
I'm not in favor of such broad changes.
This creates confusion in the end since we don't actually know if DAC driver works as is for F1 series.
I prefer we don't add blindly code for things that have not been verified.
Please restrict the changes to the part you've been able to test.